### PR TITLE
 Add a new interface to parse different types binary docker logs in one buffer

### PR DESCRIPTION
**Pull request by wodeshijie33941**
StdCopyMix is a modified version of io.Copy and it is different from StdCopy.
StdCopyMix is used for deserializing binary dockerlogs

**- What I did**
When a project needs to display both stdout and stderr type logs on the same page, the use of StdCopy functions can cause Merge logs trouble . When the log is too much, it will be a disaster. So provide a function parsed ordered log to avoid merge logs.
**- How I did it**
When parsing binary docker logs , do not distinguish stdType.
All logs are output to one buffer.
**- How to verify it**
`
	
	package test
	
	import (
		"bytes"
		"fmt"
		"github.com/docker/docker/api/types"
		"github.com/docker/docker/pkg/stdcopy"
		"golang.org/x/net/context"
		"testing"
		"time"
		"github.com/docker/docker/client"
	)
	
	// test StdCopyMix interface
	func TestStdCopyMix(t *testing.T) {
	
		// replace "675aa48d267a" with your real container id
		var ContainerID = "675aa48d267a"
		ctx := context.Background()
		cli, err := client.NewClientWithOpts(client.FromEnv)
		if err != nil {
			return
		}
		cli.NegotiateAPIVersion(ctx)
		spacetime,parseerr:= time.ParseDuration("-20m")
		untilspace,parseerr:= time.ParseDuration("-10m")
		if nil != parseerr{
			return
		}
		destTime := time.Now().Add(spacetime)
		untilTime := time.Now().Add(untilspace)
	
		temOption := types.ContainerLogsOptions{
			ShowStdout:true,
			ShowStderr:true,
			Since: destTime.Format(time.RFC3339),
			Until: untilTime.Format(time.RFC3339),
			Timestamps:false,
			Follow:false ,
			Tail:"",
			Details: false,
		}
	
		content, err := cli.ContainerLogs(ctx, ContainerID,temOption)
		if  nil != err{
			fmt.Println(err)
			return
		}
		stdout := new(bytes.Buffer)
	
		// write all logs to stdout
		_, err = stdcopy.StdCopyMix(stdout, content)
		if err != nil {
			return
		}
		fmt.Println(stdout.String())
	}	
`

- Description for the changelog

Add a new interface to parse binary docker logs
StdCopyMix is a modified version of io.Copy and it is different from StdCopy.
StdCopyMix is used for deserializing binary dockerlogs`
**- Description for the changelog**

 Add an interface to read different types of logs

**- A picture of a cute animal (not mandatory but encouraged)**

lovelybee

### CODE REVIEW

File: pkg/stdcopy/stdcopy.go [hunk 1]
```diff
@@ -188,3 +188,91 @@ func StdCopy(dstout, dsterr io.Writer, src io.Reader) (written int64, err error)
 		nr -= frameSize + stdWriterPrefixLen
```
Comment (wodeshijie33941) [inline pkg/stdcopy/stdcopy.go:188]: Add a new interface to parse binary docker logs into one buffer

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
> When a project needs to display both stdout and stderr type logs on the same page, the use of StdCopy functions can cause Merge logs trouble . When the log is too much, it will be a disaster. 

Could you be more specific?
Also, please try to minimize code duplication across `StdCopy` and `StdCopyMix`.

**Comment by thaJeztah** [comment]
Would you get the same result by using `StdCopy`, but instead of passing `stdout`  and `stderr`, passing the same `io.Writer` twice (same writer as `dstout` and `dsterr`) ?

```
func StdCopy(dstout, dsterr io.Writer, src io.Reader) (written int64, err error) {
```

**Comment by wodeshijie33941** [comment]
I called the StdCopy the way  thaJeztah said, and I got the same result.
I didn't think I could call a function like this

**Comment by thaJeztah** [comment]
Good to hear that worked 👍

**Pull request closed without merging**